### PR TITLE
Add Expr.TakeFromOutput, builder methods, and unit tests

### PR DESCRIPTION
**Pull request by jeffmay**
- Add `Expr.TakeFromOutput`
- Add `.headOption` and `.take` operators to the builder
- Add `Window.empty`
- Add unit tests

### CODE REVIEW

File: core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/TakeFromOutputSpec.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.time.Instant
+
+class TakeFromOutputSpec extends AnyWordSpec {
+
+  "Expr.TakeFromOutput" when {
+    val updateABC = FactTypes.TagsUpdate(TagsUpdate(Set("A", "B", "C"), Instant.now().minusSeconds(60 * 60 * 24)))
+    val updateDEF = FactTypes.TagsUpdate(TagsUpdate(Set("D", "E", "F"), Instant.now()))
+
+    "using .take(n) with a positive number" should {
```
Comment (jeffmay) [inline core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/TakeFromOutputSpec.scala:16]: BLKR: I need to test the 0 case and the negative case.

**Review by darmbrus — APPROVED** [review]

**Review by paul-a-kennedy-rally — COMMENTED** [review]
Looks good so far....  tag me for re-review when you've fixed your BLKR?

**Review by paul-a-kennedy-rally — APPROVED** [review]
LGTM
